manager/src/test/java/com/redhat/service/bridge/manager/workers/resources/AbstractWorkerTest.java

```diff
@@ -53,4 +77,49 @@ void workIsCompletedWhenResourceIsNotFound() {
         verify(workManager).complete(work);
     }
 
+    @Test
+    @Transactional
+    void workIsCompletedWhenMaxRetriedExceeded() {
```

Comment:
Are we sure this test is failing due to max attempts exceeded?

Yes, I know, we know it right now, but in future, if someone touch the code?

Reading the test, I do not see assertion of `areRetriesExceeded`. Really new to codebase, so just asking what is your opinion.

I didn't go deep into the code. If there is no place we throw a "specialized" exception, not a generic one (LimitExceededException for example), or we do not set "specialized" state (MAX_ATTEMPTS_EXCEEDED instead of FAILED)

Reply:
Hi @jomarko we can be reasonably comfortable that the test passes because the `Work` `FAILED` due to the maximum number of re-tries being exceeded by the set-up of the test itself. We do not have a specialized exception (indeed there are no exceptions thrown for this scenario, or the other of _timeout_) as we check state in code and do not need an exception. We also do not distinguish between `FAILED` for X or Y or Z. Just that it `FAILED`.